rclpy/src/rclpy/_rclpy.c

```diff
@@ -834,7 +830,7 @@ rclpy_destroy_node_entity(PyObject * Py_UNUSED(self), PyObject * args)
   }
 
   rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pynode, NULL);
-  rcl_ret_t ret;
+  rcl_ret_t ret = RCL_RET_ERROR;
```

Comment:
I think I'm using `cppcheck` (under the hood), but it might be using my include path to expand the macros and see that it is actually a return statement.

A compromise would be to set it in the else case, which would be correct in both checks.